helm/hpcc/values.schema.json

```diff
@@ -55,6 +55,31 @@
       },
       "additionalProperties": false
     },
+    "vaults": {
+      "description": "configuration for vaults accessed by the components",
+      "type": "object",
+      "properties": {
+        "timeout": {
+          "type": "integer"
+        },
+        "storage": {
+          "$ref": "#/definitions/vaultCategory"
+        },
+        "esp": {
+          "$ref": "#/definitions/vaultCategory"
+        },
+        "ecl": {
+          "$ref": "#/definitions/vaultCategory"
+        },
+        "ecl-user": {
```

Comment:
We should consider publishing the k8s secrets in subdirectories corresponding to the category - I think that would avoid the issue.

Reply:
The categories allow the various types of secrets to be organized and secured differently. To prevent leaking and usage for unintended purposes.  I haven't made use of "ecl-user" yet, but that is going to be the least secure category, ecl code will have direct access to the contents of the secret.  For example, if I need a key for use with some plugin or to pass in some non standard way to a 3rd party.  Because the ecl can see the contents it could return it, write it, or send it anywhere.  I don't want gateway or storage plane credentials to be accessible in that way.

When it comes to the organization of the actual vault it could be a simple as:
http://${env.VAULT_SERVICE_HOST}:${env.VAULT_SERVICE_PORT}/v1/secret/data/storage/${secret}
http://${env.VAULT_SERVICE_HOST}:${env.VAULT_SERVICE_PORT}/v1/secret/data/ecl/${secret}
http://${env.VAULT_SERVICE_HOST}:${env.VAULT_SERVICE_PORT}/v1/secret/data/ecl-user/${secret}

But those different sections can have different access rights, and allow the secrets to be organized based on how and where they are used.

Over time we may want to segment the vault access by components.  So certain roxies can call some gateways but hthor can't for example.  But I'd rather get a feel for how these things are used in the wild a bit first.  Our users can give us some feedback on how they need to further restrict.

I definitely agree on the idea of the k8s secret subdirectories.  That might mean I no longer have to mangle the names of the HTTP-CONNECT secrets.  That was for example to prevent them being used in the way the ecl-user category will be in the future.